driver/driver.go

```diff
@@ -58,6 +58,7 @@ type Driver struct {
 	nodeId   string
 	region   string
 	doTag    string
+	hasToken bool
```

Comment:
How about calling this `isController`? I think that would make the code a bit easier to understand and reduce the degree of comments we need below.

Reply:
Make sense, pushed some updates. I'll squash afterwards 👍